From jacmet@sunsite.dk  Tue Sep 23 15:58:17 2008
From: Peter Korsgaard <jacmet@sunsite.dk>
Date: Tue, 23 Sep 2008 22:01:27 +0200
Subject: Driver core: make bus_find_device_by_name() more robust
To: gregkh@suse.de, dbrownell@users.sourceforge.net, linux-kernel@vger.kernel.org
Cc: Peter Korsgaard <jacmet@sunsite.dk>
Message-ID: <1222200087-11533-1-git-send-email-jacmet@sunsite.dk>

From: Peter Korsgaard <jacmet@sunsite.dk>

Use sysfs_streq() in bus_find_device_by_name() so trailing newlines
are ignored (E.G. in bind/unbind).

Signed-off-by: Peter Korsgaard <jacmet@sunsite.dk>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/base/bus.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/drivers/base/bus.c
+++ b/drivers/base/bus.c
@@ -333,9 +333,7 @@ static int match_name(struct device *dev
 {
 	const char *name = data;
 
-	if (strcmp(name, dev->bus_id) == 0)
-		return 1;
-	return 0;
+	return sysfs_streq(name, dev->bus_id);
 }
 
 /**
